### ISSUE TITLE
Segfault when trying to download picture from invalid URL

### ISSUE THREAD

**Issue by dengelt** [vinszent/gnome-twitch#203]
This is the backtrace:

```
#0  0x000056314a57c455 in utils_download_picture_if_newer (soup=0x56314c492140 [SoupSession], url=0x7f45e804fa10 "Px\a\350E\177", timestamp=1474949189) at ../src/utils.c:128
        msg = <optimized out>
        soup_status = <optimized out>
        last_modified = <optimized out>
        ret = <optimized out>
#1  0x000056314a56715b in gt_twitch_download_picture (self=<optimized out>, url=<optimized out>, timestamp=<optimized out>)
    at ../src/gt-twitch.c:1148
#2  0x000056314a573ba3 in update_func (data=0x56314c8dcda0, udata=<optimized out>) at ../src/gt-game.c:75
        pic = <optimized out>
#3  0x00007f463a9dfd1e in g_thread_pool_thread_proxy (data=<optimized out>) at ././glib/gthreadpool.c:307
        pool = 0x7f45e8027ad0
#4  0x00007f463a9df325 in g_thread_proxy (data=0x56314c567e30) at ././glib/gthread.c:784
        thread = 0x56314c567e30
#5  0x00007f4637414464 in start_thread (arg=0x7f45c1df3700) at pthread_create.c:333
        __res = <optimized out>
        pd = 0x7f45c1df3700
        now = <optimized out>
        unwind_buf = 
              {cancel_jmp_buf = {{jmp_buf = {139937582102272, 5350997232981309276, 0, 139938457712943, 0, 139939633905728, -5419787346483415204, -5418145759198906532}, mask_was_saved = 0}}, priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
        __PRETTY_FUNCTION__ = "start_thread"
#6  0x00007f46371579df in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:105
```

Looking at the code it seems that `soup_message_new(SOUP_METHOD_HEAD, url)` returned NULL because the given URL "Px\a\350E\177" (wherever that came from) was invalid.

**Comment by vinszent** [comment]
Does this happen often? Looks like a dangling pointer.

**Comment by dengelt** [comment]
I don't think it happens often. Since I applied the fix for #155 I only had 2 segfaults, one is this one and the other consists only of gtk function calls.
